pkg/apis/network.harvester.cattle.io/v1alpha1/clusternetwork.go

```diff
@@ -0,0 +1,23 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=cn;cns,scope=Namespaced
+// +kubebuilder:printcolumn:name="DESCRIPTION",type=string,JSONPath=`.spec.description`
+// +kubebuilder:printcolumn:name="ENABLE",type=boolean,JSONPath=`.enable`
+
+type ClusterNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +optional
+	Description string `json:"description,omitempty"`
+	Enable      bool   `json:"enable"`
+	// +optional
+	Config map[string]string `json:"config,omitempty"`
```

Comment:
Can this map structure support complex configurations? For example how do we define NetConf.Capabilities?

Reply:
No, it can't.